library/src/webapp/js/lang-datepicker/lang-datepicker.js

```diff
@@ -1592,22 +1627,17 @@ $.datepicker._gotoToday = function (id) {
 						$(input).val(stringDay);
 					}
 				}
+				// Set hidden field to null if allowed
+                                if (options.allowEmptyDate && $(input).val() == "") {
```

Comment:
I'd clean up the wrapping of input here. We can just wrap it once in the setBlur function.